File: src/main/webapp/src/app/models/audio-bus.ts
import { AudioDevice } from "./audio-device";

export class AudioBus {
  uuid: string;
  audioDevice: AudioDevice;
  name: string = "";
  channels: number = 2;

  constructor(data?: any) {
    if (!data) {
      return;
    }

    this.uuid = data.uuid;

    if (data.audioDevice) {
      this.audioDevice = new AudioDevice(data.audioDevice);
    }

    this.name = data.name;
    this.channels = data.channels;
  }

}

File: src/main/java/com/ascargon/rocketshow/audio/AudioBus.java
package com.ascargon.rocketshow.audio;

import jakarta.xml.bind.annotation.XmlRootElement;
import lombok.Getter;
import lombok.Setter;

import java.util.UUID;

/**
 * An audio bus containing a number of channels.
 *
 * @author Moritz A. Vieli
 */
@XmlRootElement
@Getter
@Setter
public class AudioBus {

    private String uuid = String.valueOf(UUID.randomUUID());
    private AudioDevice audioDevice;
    private String name;
    private int channels = 2;

}
